.github/workflows/node.js.yml

```diff
@@ -8,22 +8,77 @@ on:
   workflow_call:
 
 jobs:
+  
   build:
-
     runs-on: ubuntu-latest
-
+    outputs:
+      coverage: ${{ steps.coverage.outputs.coverage }}
+      color: ${{ steps.coverage.outputs.color }}
     strategy:
       matrix:
         node-version: [16.x, 18.x]
         # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-
     steps:
     - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
       uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
         cache: 'npm'
+    - name: Checkout code
+      uses: actions/checkout@v2
     - run: npm ci
     - run: npm run fmt:check
-    - run: npm run test:ci
+    - id: coverage
+      run: |
+        value=$(npm run test:ci | grep -oE '^│\s*total\s+\│\s*([0-9.]+)%' | grep -oE '[0-9]+(\.[0-9]+)?%' | cut -d'.' -f1)
+        echo "coverage=$value" >> $GITHUB_OUTPUT
+
+    - name: Update README
+      uses: actions/github-script@v3
+      with:
+        script: |
+          let color = "red";
+
+          const coverage = parseInt(process.env.COVERAGE);
+
+          if (coverage >= 80) {
+              color = "green";
+          } else if (coverage >= 70) {
+              color = "orange";
+          }
+
+          const fs = require('fs');
+          
+          const content = fs.readFileSync('README.md', 'utf8');
+          const lines = content.split('\n');
+
+          const match = lines[1].match(/coverage-(\d{1,3})%/);
+          const oldCoverage = match ? parseInt(match[1]) : null;
+
+          if (oldCoverage != coverage || !oldCoverage) {
+            console.log('updating badge');
```

Comment:
does this log really necessary or just for debugging?

Reply:
It's for debugging, I chose to leave it. Could be useful if there is something wrong